packages/suite/src/components/wallet/WalletLayout/AccountTopPanel/AccountNavigation.tsx

```diff
@@ -1,32 +1,21 @@
 import { getNetworkOptional } from '@suite-common/wallet-config';
 import { hasNetworkFeatures } from '@suite-common/wallet-utils';
-import { EventType, analytics } from '@trezor/suite-analytics';
 
-import { goto } from 'src/actions/suite/routerActions';
 import { Translation } from 'src/components/suite/Translation';
 import { NavigationItem, SubpageNavigation } from 'src/components/suite/layouts/SuiteLayout';
-import { useDispatch, useSelector } from 'src/hooks/suite';
+import { useGoToWithAnalytics } from 'src/components/suite/layouts/SuiteLayout/PageHeader/useGoToWithAnalytics';
+import { useSelector } from 'src/hooks/suite';
 import { selectHasExperimentalFeature } from 'src/reducers/suite/suiteReducer';
 import { selectSelectedAccount } from 'src/reducers/wallet/selectedAccountReducer';
 import { WalletParams } from 'src/types/wallet';
 
 export const AccountNavigation = () => {
     const account = useSelector(selectSelectedAccount);
     const routerParams = useSelector(state => state.router.params) as WalletParams;
-    const dispatch = useDispatch();
     const enabledNftSection = useSelector(selectHasExperimentalFeature('nft-section'));
     const network = getNetworkOptional(routerParams?.symbol);
     const networkType = account?.networkType || network?.networkType || '';
-
-    const goToWithAnalytics = (...[routeName, options]: Parameters<typeof goto>) => {
-        if (account?.symbol) {
-            analytics.report({
-                type: EventType.AccountsActions,
-                payload: { symbol: account.symbol, action: routeName },
-            });
-        }
-        dispatch(goto(routeName, options));
-    };
+    const goToWithAnalytics = useGoToWithAnalytics();
```

Comment:
```suggestion
    const goToWithAnalytics = useGoToWithAnalytics(account);
```
In this case it can be propagated from here right?

Reply:
yes, my bad. updated